airflow/api_fastapi/core_api/routes/public/task_instances.py

```diff
@@ -322,7 +333,85 @@ def get_task_instances(
         session,
     )
 
-    task_instances = session.scalars(task_instance_select).all()
+    task_instances = session.scalars(task_instance_select)
+
+    return TaskInstanceCollectionResponse(
+        task_instances=[
+            TaskInstanceResponse.model_validate(task_instance, from_attributes=True)
+            for task_instance in task_instances
+        ],
+        total_entries=total_entries,
+    )
+
+
+@task_instances_router.post(
+    "/list",
```

Comment:
@pierrejeambrun , wouldn't this endpoint be wrongly invoked when user makes a call to `/dags/<Random string>/dagRuns/<Random string>/taskInstances/list` ?

Reply:
Yes I think we need to manually constraint those. (It was enforced by connexion before, we need it done in pydantic as well).

PR here https://github.com/apache/airflow/pull/44175

Thanks